llvm/lib/Target/SystemZ/SystemZISelLowering.cpp

```diff
@@ -1440,9 +1440,11 @@ bool SystemZTargetLowering::findOptimalMemOpLowering(
                                                   SrcAS, FuncAttributes);
 }
 
-EVT SystemZTargetLowering::getOptimalMemOpType(const MemOp &Op,
-                                   const AttributeList &FuncAttributes) const {
-  return Subtarget.hasVector() ? MVT::v2i64 : MVT::Other;
+EVT SystemZTargetLowering::getOptimalMemOpType(
+    const MemOp &Op, const AttributeList &FuncAttributes) const {
+  return Subtarget.hasVector() && !useIntScalarMemOps(FuncAttributes)
```

Comment:
Not sure why this is necessary.  This is supposed to handled once and for all in `SystemZSubtarget::initializeSubtargetDependencies` by switching off the vector facility completely for soft-float:
```
  // -msoft-float implies -mno-vx.
  if (HasSoftFloat)
    HasVector = false;
```

Reply:
I think you might be right that it's not needed. I don't actually recall why I modified this function in the initial version of the PR.

I'll re-check and remove it if it's unnecessary.